test/Serilog.Tests/Settings/KeyValuePairSettingsTests.cs

```diff
@@ -124,6 +125,37 @@ public void SinksAreConfigured()
             Assert.Empty(DummyRollingFileAuditSink.Emitted);
         }
 
+        [Fact]
+        public void WritingToUndeclaredSinkWritesToSelfLog()
+        {
+            List<string> messages = new();
```

Comment:
Unfortunately, because the tests run in parallel, this will collide with others. Right now, I think the best workaround would be to move this one test over to `SelfLogTests`, add:

```csharp
            Messages = new();
            SelfLog.Enable(m =>
            {
                Messages = Messages ?? new();
                Messages.Add(m);
            });
```

instead of this block, and then remove the `SelfLog.Disable()` calls from this and the other test in that class.

Longer term, we should be able to move this to a test helper 🤔

Reply:
Thanks for catching that. The new test does seem to fit in better under `SelfLogTests`. I don't think I can remove `SelfLog.Disable()` in the existing test though. It relies on that method call to do an additional test -- that nothing is written to `SelfLog` when it is disabled. Should I proceed to remove `SelfLog.Disable()` in the new test but keep the existing one as is?

Reply:
Looks like adding xUnit.net's `[CollectionDefinition("SelfLog", DisableTestParallelization = true)]` to the class would allow us to keep the `Disable()` calls in there.

Reply:
It looks like parallelization is already disabled on the entire assembly: https://github.com/serilog/serilog/blob/dev/test/Serilog.Tests/Properties/AssemblyInfo.cs. When I run `./build.sh`, I get all tests passing.